### PR TITLE
change enqueue to optional action

### PR DESCRIPTION
**Pull request by wpeng102**
Signed-off-by: wpeng102 <wpeng102@126.com>

The enqueue action is a mandatory action right now. It would be wonderful to make it optional and configurable.
Firstly the integration code with other community will be more simple and easy.
Secondly the other actions no need to depend on enqueue . All the actions are decoupled.

https://github.com/volcano-sh/volcano/issues/2040

If not config `Enqueue` action, scheduler will change the `Pending` pg to `Inqueue` at closesession.

### CODE REVIEW

File: pkg/scheduler/framework/job_updater.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ import (
 	"volcano.sh/volcano/pkg/scheduler/api"
 )
 
+var EnqueueExist bool
```
Comment (Thor-wl) [inline pkg/scheduler/framework/job_updater.go:16]: Kindly to ask about why hard code enqueue action to make the judgement?

Comment (wpeng102) [reply pkg/scheduler/framework/job_updater.go]: Introduce ActionMap to store which action is configured by customer.

**Comment by wpeng102** [comment]
Backgroud:
Some customers wants volcano do not block pod created when resource is insufficient, just keep same behavior as default scheduler.

File: pkg/scheduler/conf/scheduler_conf.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ limitations under the License.
 
 package conf
 
+// ActionMap check if a action exist in scheduler configmap
+var ActionMap map[string]bool
```
Comment (Thor-wl) [inline pkg/scheduler/conf/scheduler_conf.go:20]: Maybe `ExistActionMap` is better?

Comment (wpeng102) [reply pkg/scheduler/conf/scheduler_conf.go]: EnabledActionMap is better

Comment (Thor-wl) [reply pkg/scheduler/conf/scheduler_conf.go]: I'm OK about that.

File: pkg/scheduler/framework/job_updater.go [hunk 3]
```diff
@@ -98,6 +99,10 @@ func (ju *jobUpdater) updateJob(index int) {
 	job := ju.jobQueue[index]
 	ssn := ju.ssn
 
+	if !conf.EnabledActionMap["enqueue"] && job.PodGroup.Status.Phase == scheduling.PodGroupPending {
```
Comment (william-wang) [inline pkg/scheduler/framework/job_updater.go:102]: please add comments here or extract a small func for this special treatment

**Comment by william-wang** [comment]
Any e2e for scheduler configuration without `enqueu`?

File: pkg/scheduler/framework/job_updater.go (hunk 3 above)
Comment (wpeng102) [reply pkg/scheduler/framework/job_updater.go:103]: add comments for this block

**Comment by wpeng102** [comment]
> Any e2e for scheduler configuration without `enqueue`?

Add e2e case

**Review by william-wang — COMMENTED** [review]
/lgtm

**Pull request merged**